test/unit/eigensolver/mc/test_backtransformation.cpp

```diff
@@ -130,16 +130,18 @@ void testBacktransformationEigenv(SizeType m, SizeType n, SizeType mb, SizeType
       for (SizeType t = 0; t < mb && nt < tottaus; ++t) {
         const GlobalElementIndex v_offset{i * mb + t, i * mb + t};
         auto dotprod = blas::dot(m - t, mat_v_loc.ptr(v_offset), 1, mat_v_loc.ptr(v_offset), 1);
-        T taui;
+        BaseType<T> tau_i;
         if (std::is_same<T, ComplexType<T>>::value) {
           auto seed = 10000 * i + 1;
-          dlaf::matrix::util::internal::getter_random<T> random_value(seed);
-          taui = random_value();
+          dlaf::matrix::util::internal::getter_random<BaseType<T>> random_value(seed);

```

Comment:
The random number generator should be seed for each block, and not for each element.
(Currently tau_i has mb time the same value)